### PR TITLE
Add CI testing as part of push to master branch and PRs. Avoid accidental bad code merges

### PR DESCRIPTION
**Pull request by yashs360**
Test PR with CI testing

### CODE REVIEW

**Comment by yashs360** [comment]
Successful CI tests on branch push - https://github.com/yashs360/dice/actions/runs/9856408115/job/27213337521

Working on build on PR creation.

**Comment by yashs360** [comment]
Tests getting triggered on PR. Showing in-progress/success status.

Screenshot 2024-07-09 at 9 55 13 PM
Screenshot 2024-07-09 at 9 54 54 PM

**Comment by yashs360** [comment]
wip - test induced error to block merge on test failures

**Comment by yashs360** [comment]
Tested failure scenario by purposeful error induced - 

Screenshot 2024-07-09 at 9 59 08 PM

**Comment by yashs360** [comment]
Ready for review @arpitbbhayani @JyotinderSingh

**Review by JyotinderSingh — APPROVED** [review]

File: core/bloom_test.go [hunk 1]
```diff
@@ -103,8 +103,6 @@ func TestBloomFilter(t *testing.T) {
 }
 
 func TestGetOrCreateBloomFilter(t *testing.T) {
```
Comment (yashs360) [inline core/bloom_test.go:105]: Not sure why these changes are showing here. These were merged already 🤔

Comment (JyotinderSingh) [reply core/bloom_test.go:105]: You may need to pull the master into your branch.

Comment (yashs360) [reply core/bloom_test.go]: fixed now. I had to hit Sync Fork on my fork.

**Comment by arpitbbhayani** [comment]
Thanks @yashs360!
